p2p-specs/p2p-interface.md

```diff
@@ -168,22 +173,28 @@ For any optional queueing, Bundlers SHOULD maintain maximum queue sizes to avoid
 
 #### Global topics
 
-The primary global topics used to propagate user operations to all nodes on the network is `UserOperationsWithEntryPoint`.
+The primary global topics used to propagate user operations to all nodes on the network is `verified_user_operations`.
+
+##### `verified_user_operations`
+
+The `verified_user_operations` topic is used to gossip user operations on mempool topics where the user operation is valid.
 
-##### `user_ops_with_entry_point`
+The following validations MUST pass before forwarding the `verified_user_operations` on the network:
+- _[IGNORE]_ `verified_at_block_hash` is too far in the past.
+- _[REJECT]_ If any of the sanity checks specified in the [EIP](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4337.md#client-behavior-upon-receiving-a-useroperation) fails.
+- _[REJECT]_ If the simulated validation of the user operation fails.
 
-The `user_ops_with_entry_point` topic is the concatenation of EntryPoint address and a list of UserOperations corresponding to the entry point address. This message is serialized using SSZ.
+Upon receiving a user operation from an `eth_sendUserOperation` RPC call, a bundler SHOULD:
 
-The following validations MUST pass before forwarding the `user_ops_with_entry_point` on the network:
-- [IGNORE] `verified_at_block_hash` is too far in the past.
-- [REJECT] If any of the sanity checks specified in the [EIP](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4337.md#client-behavior-upon-receiving-a-useroperation) fails.
-- [REJECT] If the simulated validation of the user operation fails.
+1. Simulate the user operation to determine which mempools it is valid on.
+2. It the user operation is valid on the canonical mempool, ONLY send it on the canonical mempool topic.
+3. Else, gossip the user operation on each mempool topic for which the user operation is valid.
```

Comment:
do we make this distinction of canonical/other mempool?
consider a widely-supported USDC mempool: does it mean a UserOp is propagated differently if it uses that USDC mempool than if it is supported by the canonical one?
or do we implicitly rely on the protocol optimization that it will not send the same message to the same peer node if they share multiple topics.

Reply:
We currently don't make this distinction, and the protocol WILL send the same message to each peer on each topic they share.

We need some way to limit this traffic when a UO matches multiple shared mempools. This was an attempt to say that if a UO is valid on the cannonical mempool (i.e. doesn't use USDC), don't also send it in the USDC mempool (even though it is technically valid there).

Thoughts on how to improve this?

Reply:
We decided on the call to stick with this approach.

We will publish a list of canonical mempools somewhere in this repo. This will be an ordered list.

The bundler will traverse the canonical mempools in order, if it matches a canonical mempool it is only sent on that mempool. If no matches, the UO is sent on ALL alternative mempools that it is a match on.

This is subject to change in the future but gets us going until we have a more concrete approach to alt-mempools in the future.

I'll make the changes in the PR.